pymc3/distributions/timeseries.py

```diff
@@ -71,3 +72,52 @@ def logp(self, x):
 
         innov_like = Normal.dist(mu=x_im1 + mu, tau=tau, sd=sd).logp(x_i)
         return init.logp(x[0]) + T.sum(innov_like)
+
+
+class GARCH11(Continuous):
+    """
+    GARCH(1,1) with Normal innovations. The model is specified by
+
+    y_t = sigma_t * z_t
+    sigma_t^2 = omega + alpha_1 * y_{t-1}^2 + beta_1 * sigma_{t-1}^2
+
+    with z_t iid and Normal with mean zero and unit standard deviation.
+
+    Parameters
+    ----------
+    omega : distribution
+        omega > 0, distribution for mean variance
+    alpha_1 : distribution
+        alpha_1 >= 0, distribution for autoregressive term
+    beta_1 : distribution
+        beta_1 >= 0, alpha_1 + beta_1 < 1, distribution for moving
+        average term
+    initial_vol : distribution
+        initial_vol >= 0, distribution for initial volatility, sigma_0
+    """
+    def __init__(self, omega=None, alpha_1=None, beta_1=None,
+                 initial_vol=None, *args, **kwargs):
+        super(GARCH11, self).__init__(*args, **kwargs)
+
+        self.omega = omega
+        self.alpha_1 = alpha_1
+        self.beta_1 = beta_1
+        self.initial_vol = initial_vol
+        self.mean = 0
+
+    def _get_volatility(self, x):
+
+        def volatility_update(x, vol, w, a, b):
+            return T.sqrt(w + a * T.square(x) + b * T.square(vol))
+
+        vol, _ = scan(fn=volatility_update,
+                      sequences=[x],
+                      outputs_info=[self.initial_vol],
+                      non_sequences=[self.omega, self.alpha_1,
+                                     self.beta_1])
```

Comment:
What do you run the trace on then - I'm trying it with Metropolis and getting errors. 

I'm using 
```python 
r = shared(np.array([28, 8, -3, 7, -1, 1, 18, 12, 15], dtype=np.float32))
sigma1 = shared(np.array(.2, dtype=np.float32))
alpha0 = shared(np.array(.5, dtype=np.float32))

with Model() as garch:
    alpha1 = Normal('alpha1', 0., 1., dtype='float32')
    BoundedNormal = Bound(Normal, lower=0., upper=(1 - alpha1))
    beta1 = BoundedNormal('beta1', 0., sd=1e3, dtype='float32')

    obs = GARCH11('garchy_garch', omega=alpha0, alpha_1=alpha1,
                  beta_1=beta1, initial_vol=sigma1, observed=r,
                  dtype='float32')
    start = find_MAP(fmin=optimize.fmin_bfgs)

def run(n=1000):
    if n == "short":
        n = 50
    with garch:
        h = find_hessian(start)
        step = Metropolis(garch.vars, h, blocked=True)
        trace = sample(n, step, start)


    burn = n / 10

    traceplot(trace[burn:])
    plots.summary(trace[burn:])


if __name__ == '__main__':
    run()
```
How are you getting a traceplot or whatever?